Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm

```diff
@@ -1780,18 +1584,28 @@ static void jsPDFDocFinalize(JSObjectRef object)
     return convertFromPDFViewToRootView(IntRect(IntPoint(), size()));
 }
 
-void PDFPlugin::calculateSizes()
+IntSize PDFPlugin::contentsSize() const
 {
-    if ([m_pdfDocument isLocked]) {
-        m_firstPageHeight = 0;
-        setPDFDocumentSize(IntSize(0, 0));
-        return;
-    }
+    if (isLocked())
+        return { 0, 0 };
+    return IntSize([m_pdfLayerController contentSizeRespectingZoom]);
+}
 
-    m_firstPageHeight = [m_pdfDocument pageCount] ? static_cast<unsigned>(CGCeiling([[m_pdfDocument pageAtIndex:0] boundsForBox:kPDFDisplayBoxCropBox].size.height)) : 0;
-    setPDFDocumentSize(IntSize([m_pdfLayerController contentSizeRespectingZoom]));
+unsigned PDFPlugin::firstPageHeight() const
+{
+    if (isLocked() || ![m_pdfDocument pageCount])
+        return 0;
+    return static_cast<unsigned>(CGCeiling([[m_pdfDocument pageAtIndex:0] boundsForBox:kPDFDisplayBoxCropBox].size.height));
+}
 
-    if (!m_frame || !m_frame->page())
+bool PDFPlugin::isLocked() const
+{
+    return [m_pdfDocument isLocked];
+}
+
+void PDFPlugin::updatePDFHUDLocation()
+{
+    if (isLocked() || !m_frame || !m_frame->page())
         return;
     m_frame->page()->updatePDFHUDLocation(*this, frameForHUD());
```

Comment:
checkedPage()

Reply:
Doesn't currently exist! This is a WebFrame

Reply:
Sorry, I got confused and thought I was in WebCore for a minute  :D
`m_frame->protectedPage()->updatePDFHUDLocation(*this, frameForHUD());` given that WebPage is refcounted.